File: Abhjit/weekone/plusOne.cpp
// problem link https://leetcode.com/problems/plus-one/
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        vector<int> i;
        int t = digits.size()-1;
        int a =digits[t];
        for(int i = t;i>=0;i--){
        if(digits[i] != 9){
           digits[i]=digits[i]+1;
         return digits;
         }
         else{
            digits[i]=0;
        }
        
        
        }
        digits[0]=1;
        digits.push_back(0);
        return digits;     
    }
};

File: Srija/Week 1/plusone.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        flag = True
        i = len(digits)-1
        while (flag and i>=0):
            if digits[i]==9:
                digits[i] = 0
            else:
                digits[i] += 1
                flag = False
            i-=1
        if flag == True:
            digits.insert(0, 1)
        return digits